rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSourceUtils.java

```diff
@@ -98,8 +98,9 @@ public static void validateDbKind(DbKind currentDbKind, String databaseDir, bool
         }
 
         boolean databaseDirExists = dir.exists() && dir.isDirectory();
+        File file = new File(databaseDir, DB_KIND_PROPERTIES_FILE);
 
-        if (!databaseDirExists || dir.list().length == 0) {
+        if (!databaseDirExists || dir.list().length == 0 || !file.exists() || !file.canRead()) {
             KeyValueDataSourceUtils.generatedDbKindFile(currentDbKind, databaseDir);
```

Comment:
in regards to [this](https://github.com/rsksmart/rskj/pull/2063/files#r1260771883) comment.. we should probably have something like this?

```
    public static void validateDbKind(DbKind currentDbKind, String databaseDir) {
        File dbDir = new File(databaseDir);
        boolean dbDirExists = dbDir.exists();

        if (dbDirExists && !dbDir.isDirectory()) {
            LoggerFactory.getLogger(KEYVALUE_DATASOURCE).error("database.dir should be a folder.");
            throw new IllegalStateException("database.dir should be a folder");
        }

        File dbKindFile = new File(dbDir, DB_KIND_PROPERTIES_FILE);
        boolean dbKindFileExists = dbKindFile.exists();
        if (dbKindFileExists && !dbKindFile.isFile()) {
            LoggerFactory.getLogger(KEYVALUE_DATASOURCE).error("dbKind file should be a file.");
            throw new IllegalStateException("dbKind file should be a file");
        }

        boolean isEmptyDbDir = dbDirExists && Objects.requireNonNull(dbDir.list()).length == 0;
        if (!dbDirExists || isEmptyDbDir) { // use dbKind from config, if db folder doesn't exist or is empty
            KeyValueDataSourceUtils.generatedDbKindFile(currentDbKind, databaseDir);
            return;
        }
        if (!dbKindFileExists) { // use LEVEL_DB (for backward compatibility), if db folder is not empty and dbKind file doesn't exist
            KeyValueDataSourceUtils.generatedDbKindFile(DbKind.LEVEL_DB, databaseDir);
            return;
        }

        DbKind prevDbKind = KeyValueDataSourceUtils.getDbKindValueFromDbKindFile(databaseDir);
        if (prevDbKind != currentDbKind) {
            LoggerFactory.getLogger(KEYVALUE_DATASOURCE).warn("Current Db kind {} does not match with db kind {} from properties file, using value from properties file." +
                            " keyvalue.datasource from .conf file is used the first time the node is run or db is created from scratch.",
                    currentDbKind.name(), prevDbKind.name());
        }
    }
```
let me know what you think about this approach..

(note: I also did some optimisations just not to do multiple sys calls. Also it looks like `databaseReset` is redundant here, cause that flag would result to db folder being empty in anyway, or am I missing something?)

Reply:
That makes sense, I'll update the code.

Reply:
Done, following the reasoning I explained in your last comment, I added the check for `dkKindFile.canRead()` in the same if that checks for the [file existing](https://github.com/rsksmart/rskj/pull/2063/files#diff-a0ef9af4ae2ae4279e4a9cbb5f4d51ad3924c22a3464f433bc5fbfb694f38aa8R114).